src/Umbraco.Core/Xml/UmbracoXPathPathSyntaxParser.cs

```diff
@@ -134,11 +140,7 @@ public class UmbracoXPathPathSyntaxParser
                     "$site",
                     string.Format(rootXpath, closestPublishedAncestorId) + "/ancestor-or-self::*[@level = 1]");
             });
-        }
 
-        // These parameters must have a node id context
-        if (nodeContextId.HasValue)
```

Comment:
Won't this change make it so that if there is a `parentId` that will always be used instead of the `nodeContextId` for setting the `$current`?